TournamentTabItem pull request

NikPletnev opened the pull request.

NikPletnev committed:
added UI for Tournaments

NikPletnev committed:
Added  view model  and mapper for tournaments

NikPletnev committed:
Added View model for Tournament
Updated UI for tournament
Now UI can create, delete and update tournaments

NikPletnev committed:
Add some UI buttons and blank for Participant

NikPletnev committed:
Added first commands
Updated view model

NikPletnev committed:
Fixed DeleteTournament Command

NikPletnev committed:
completed comands for Tournaments main Tab

NikPletnev committed:
Updated commands

NikPletnev committed:
Changed tables ReusltPlayerInTournament and ResultTeam in tournament, changed storedProcedure DeeteByTournamentId
Now tables can allow null in result, round and match columns

NikPletnev committed:
added orject for tests

NikPletnev committed:
Added project for tests and tests for the tournament
Changed tournament, added OnlyForTeams property, created interfaces for repositories. Tournaments are now initialized and created to be only for teams or players.

NikPletnev committed:
preparations for working with team tournaments

NikPletnev committed:
changed connection string

Negodyaj commented on TournamentOrganaizer.DataLayer/Repositories/ResultTournamentPlayerRepository.cs:
```diff
@@ -132,18 +132,56 @@ public void SetPlayerResultInRoundOfTournament(int newResult, ResultTournamentPl
+                    new
+                    {
+                        TournamentId = tournamentId,
+                        Id = id
```
а Id - это чей айди?..

NikPletnev replied:
Id - это id игрока которого удаляют из определенного турнира
Переименовал в playerId

Negodyaj commented on TournamentOrganaizer.DataLayer/Repositories/TournamentRepository.cs:
```diff
@@ -11,23 +11,25 @@
             const string procedureName = "Tournament_Insert";
             using IDbConnection connection = ProvideConnection();
-            connection.Execute(
+            var result = connection.Query<int>(
```
по-моему, из этой хранимки айдишник не возвращается
ну или ты забыл поменять текст хранимки в дб-проекте

NikPletnev replied:
Да, я менял хранимку, но пока не публиковал Дб на сервере 
На своем тестовом я все переписал
Пока не публикую, так как ребята работают со старой дб и  я думал публиковать, когда уже все смерджиться в мастер

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+            Assert.IsNotNull(actual[0].Game);
+            Assert.IsInstanceOf(typeof(TournamentModel), actual[0]);
+            Assert.IsInstanceOf(typeof(GameModel), actual[0].Game);
+            Assert.Pass();
```
Assert.Pass() - найс!

NikPletnev replied:
Почему нет?) 
В целом могу удалить, он не мешает

NikPletnev replied:
Убрал

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
```
а разные тесткейсы хоть что-то меняют?..

NikPletnev replied:
Нет.

NikPletnev replied:
Оставил 1 тест кейс

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+        }
+
+        [TestCase(1)]
+        [TestCase(2)]
```
ну и далее по тексту - тоже

NikPletnev replied:
+

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+        [TestCase(1)]
+        [TestCase(2)]
+
+
```
ненужные пустые строки стоит убирать

NikPletnev replied:
+

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+            sut.UpdateTournament(tetsTournament);
+
+            //assert
+            Assert.Pass();
```
збс)))

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+        [TestCase(2)]
+        [TestCase(3)]
+
+        public void DeleteTournamentTest_ShouldPass(int id)
```
поубирай слово Test из названий методов

NikPletnev replied:
+

Negodyaj commented on TournamentOrganizer.BusinessLayer.Tests/TournamentServiceTests.cs:
```diff
@@ -0,0 +1,362 @@
+            var sut = new TournamentService(_tournamnetRepositoryMock.Object, _gameRepositoryMock.Object, _resultTournamentPlayerRepositoryMock.Object);
+
+            //act
+            var actual = sut.InsertTournament(testTournament);
```
а вместе с турниром никакие другие сущности не добавляются?

NikPletnev replied:
Нет, тут создаются только турниры
Игры к турнирам подвязываются уже существующие так же как и игроки

Negodyaj commented on TournamentOrganizer.BusinessLayer/Models/GameModel.cs:
```diff
@@ -0,0 +1,43 @@
+            set
+            {
+                _id = value;
+                OnPropertyChanged(nameof(Id));
```
а почему OnPropertyChanged зовётся на уровне бизнес-логики? (тут и далее)
это штуки-дрюки UI-я

Negodyaj commented on TournamentOrganizer.BusinessLayer/Service/TournamentService.cs:
```diff
@@ -0,0 +1,86 @@
+
+        public void DeleteTournament(int id)
+        {
+            _tournamentRepository.TournamentDeleteById(id);
```
а в каком случае нам реально надо удалять турнир?..
а что будет, если внешний ключ ругнётся?

NikPletnev replied:
Решил что такую возможно нужно предоставить пользователю 
Еще не добавил валидацию для создания турниров 
Турниры можно будет удалить только если он пустой без матчей и игроков

Negodyaj commented on TournamentOrganizer/VeiwModels/TabItemTournamentsViewModel.cs:
```diff
@@ -0,0 +1,269 @@
+            }
+        }
+
+        public TabItemTournamentsViewModel()
```
где должен находиться конструктор класса? между чем и чем?)

NikPletnev replied:
Между свойствами-полями и методами

Negodyaj commented on TournamentOrganizer.BusinessLayer/Models/IParticipant.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace TournamentOrganizer.BusinessLayer.Models
+{
+    public interface IParticipant 
```
а почему это интерфейс? а зачем он нужен? а какие классы должны иметь какое-то к нему отношение?

NikPletnev replied:
Этот интерфейс еще не используется в полную силу
Пока турниры работают только с игроками 
Потом его допишу

NikPletnev committed:
Added ImprovedObservableCollection, now tournaments correctly works with teams
Changed stored procedures, repositories and tournament service
Fixed bugs according coments

NikPletnev committed:
Fixed major bugs with teams and procedures

NikPletnev committed:
Test update

NikPletnev committed:
Tests updated

NikPletnev committed:
added text validation

NikPletnev committed:
Changed connection string

NikPletnev committed:
Merge branch 'master' into TournamentTabItem
